x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md

```diff
@@ -176,3 +176,44 @@ row a = [true, false, false, true]
 ```
 
 That skips nodes that don't have the `esql.mv_sort` feature.
+
+
+### Warnings
+
+Some queries can return warnings, eg. for number overflows or when a multi-value is passed to a funciton
+that does not support it.
+
+Each CSV-SPEC test has to also assert all the expected warnings.
+
+Warnings can be specified as plain text or as a regular expression (but a single test cannot have a mix of both).
+Each warning has to be specified on a single row, between the query and the result, prefixed by `warning:` or `warningRegex:`.
+If multiple warnings are defined, the order is not relevant.
+
+This is an example of how to test a query that returns two warnings:
+
+```csv-spec
+addLongOverflow
+row max = 9223372036854775807 | eval sum = max + 1 | keep sum;
+
+warning:Line 1:44: evaluation of [max + 1] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 1:44: java.lang.ArithmeticException: long overflow
+
+sum:long
+null
+;
+```
+
+The same, using regular expressions:
+
+```csv-spec
+addLongOverflow
+row max = 9223372036854775807 | eval sum = max + 1 | keep sum;
+
+warningRegex:Line 1:44: evaluation of \[max + 1\] failed, treating result as null. Only first 20 failures recorded.
```

Comment:
Should we update the example to actually contain a regex? Maybe smth like the line location (`..Line \d+:\d+..`), even though that doesn't actually vary.